ui/components/component-library/tag-url/tag-url.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Box from '../../ui/box/box';
+import { Text } from '../text';
+import {
+  ALIGN_ITEMS,
+  BORDER_RADIUS,
+  COLORS,
+  DISPLAY,
+  SIZES,
+  TEXT,
+} from '../../../helpers/constants/design-system';
+import { AvatarFavicon } from '../avatar-favicon';
+import { ButtonLink } from '../button-link';
+import { Icon, ICON_NAMES } from '../icon';
+
+export const TagUrl = ({
+  label,
+  labelProps,
+  actionButtonLabel,
+  actionButtonProps,
+  src,
+  avatarFaviconProps,
+  fallbackIconProps,
+  className,
+  ...props
+}) => {
+  return (
+    <Box
+      className={classnames('mm-tag-url', className)}
+      backgroundColor={COLORS.BACKGROUND_DEFAULT}
+      borderColor={COLORS.BORDER_DEFAULT}
+      borderWidth={1}
+      alignItems={ALIGN_ITEMS.CENTER}
+      paddingLeft={2}
+      paddingRight={4}
+      gap={2}
+      borderRadius={BORDER_RADIUS.PILL}
+      display={DISPLAY.INLINE_FLEX}
+      {...props}
+    >
+      <AvatarFavicon imageSource={src} {...avatarFaviconProps} />
+      <Icon
+        className="tag-url__lock-icon"
+        name={ICON_NAMES.LOCK_FILLED}
+        color={COLORS.ICON_DEFAULT}
+        size={SIZES.SM}
+        aria-label="https://"
+        {...fallbackIconProps}
+      />
```

Comment:
I think we want to make the display of this conditional see this Figma comment thread https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C?node-id=10002:38649#295155189

So maybe something like `showLockIcon` or something. Also I'm not sure `fallbackIconProps` is a suitable prop name?